eng/common/TestResources/New-TestResources.ps1

```diff
@@ -249,6 +250,7 @@ function BuildDeploymentOutputs([string]$serviceDirectoryPrefix, [object]$azCont
         "${serviceDirectoryPrefix}AZURE_AUTHORITY_HOST" = $azContext.Environment.ActiveDirectoryAuthority;
         "${serviceDirectoryPrefix}RESOURCE_MANAGER_URL" = $azContext.Environment.ResourceManagerUrl;
         "${serviceDirectoryPrefix}SERVICE_MANAGEMENT_URL" = $azContext.Environment.ServiceManagementUrl;
+        "AZURE_SERVICE_DIRECTORY" = $serviceName.ToUpperInvariant();
```

Comment:
Given you're doing most (perf-wise) of what `BuildServiceDirectoryPrefix` is doing here, why not just do it all in this scope? Calling a separate function just to do this again + concat "_" is wasteful.

Reply:
Yeah this was annoying. It's used in two places, so I put it in a function to make sure the value was consistent (whereas before it was defined in the same scope). So the alternative was either: hardcoding the convention twice, trimming the underscore off the string, removing the `ToUpper` call from the inner function, or returning two values from the function. I don't think any option is particularly good, so maybe just hardcoding is simplest.